### ISSUE TITLE
`StatusBar.setOverlaysWebView({ overlay: true })` alongside SplashScreen glitches on Android API 33 (Android 13)

### ISSUE THREAD

**Issue by jbouchard24** [ionic-team/capacitor-plugins#1160]
### Plugin(s)

- @capacitor/splash-screen@4.0.1
- @capacitor/status-bar@4.0.1

### Capacitor Version

```
   Capacitor Doctor   

Latest Dependencies:

  @capacitor/cli: 4.1.0
  @capacitor/core: 4.1.0
  @capacitor/android: 4.1.0
  @capacitor/ios: 4.1.0

Installed Dependencies:

  @capacitor/ios: not installed
  @capacitor/cli: 4.1.0
  @capacitor/android: 4.1.0
  @capacitor/core: 4.1.0

[success] Android looking great! 👌
```

### Platform(s)

- @capacitor/android@4.1.0
- Android API 33

### Current Behavior
I have an app that uses the [`StatusBar.setOverlaysWebView({ overlay: true })`](https://capacitorjs.com/docs/apis/status-bar#setoverlayswebview) feature from the StatusBar plugin, alongside the SplashScreen plugin. When opening the app on Android API 33 (Android 13) from home screen or app drawer, the status bar is opaque and the webview stretches underneath the navigation/gesture bar. That is not the case on Android API 31 (Android 12) or earlier, or when opening the app from settings > Apps or long-pressing the app icon > App Info > Open. (See **Additional context** for more info)

Current behavior on Android API 33 (Android 13):
<img width="368" alt="image" src="https://user-images.githubusercontent.com/54321952/187204935-30212e18-0a8a-404b-b415-f84899b1e2a3.png">

### Expected Behavior
I expect a behavior like Android API 31 (Android 12) and earlier, where the app is using the space underneath a transparent status bar and the bottom doesn't stretch past the gesture bar when calling `StatusBar.setOverlaysWebView({ overlay: true })`:
<img width="368" alt="image" src="https://user-images.githubusercontent.com/54321952/187222480-8c38565a-fc41-4a93-9e54-f23ed6ce8044.png">

### Code Reproduction
[Example repo](https://github.com/jbouchard24/ionic6-capacitor4-testsplash)

### Additional Context
Here some of the tests I made. Hopefully it helps isolate the issue.

Working: ✅
- Android API 31 and earlier
- StatusBar plugin alone (no SplashScreen plugin)
- Android API 33 when long-pressing the app icon > App Info > Open

Not working: ❌
- Android API 33 when opening from home screen or app drawer

**Comment by Ionitron** [comment]
This issue may need more information before it can be addressed. In particular, it will need a reliable Code Reproduction that demonstrates the issue.

Please see the Contributing Guide for [how to create a Code Reproduction](https://github.com/ionic-team/capacitor/blob/HEAD/CONTRIBUTING.md#creating-a-code-reproduction).

Thanks!
Ionitron 💙

**Comment by jbouchard24** [comment]
Code Repro has been updated from a list of steps to an example repo

**Comment by nikitalpopov** [comment]
@jbouchard24 I've got the same issue and [this fix](https://www.anycodings.com/questions/getting-warning-application-should-not-provide-its-own-splash-screen-in-android-studio) seems to work in this case. This part in particular:

**MainActivity.kt:**

```kotlin
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen

class MainActivity : BridgeActivity() {
  public override fun onCreate(savedInstanceState: Bundle?) {
    registerPlugins(
      // ...
    )
    super.onCreate(savedInstanceState)
    // This line fixes the issue:
    val splashScreen = installSplashScreen()
  }
}
```

**Comment by jbouchard24** [comment]
@nikitalpopov Thanks for the reply, but it's not working in my project.

I have been suggested to make changes to the status bar in a `setTimeout` after hiding the splash screen. I don't like the fix, but at least it's working in my case:

```
await SplashScreen.hide();
setTimeout(async () => {
    await StatusBar.setOverlaysWebView({ overlay: true });
}, 500);
```

**Comment by hellokingdom** [comment]
@jcesarmobile we are having the same issue and it's causing quite visual JANK now as the webview is re-sized once the splash has been removed which causes a complete re-layout of the application.

Do you know if it's possible to force the webview to overlay the status bar by default without using the StatsBar plugin?

**Comment by hellokingdom** [comment]
@jcesarmobile is it possible to set-up a Capacitor project so that 

`StatusBar.setOverlaysWebView({ overlay: true });`

Is enabled by default, for example by using capacitor.config.ts

`  plugins: {
    SplashScreen: {
      ...
    },
    StatusBar: {
      overlay: true
    }
  },`

**Comment by mpmua** [comment]
> setTimeout(async () => {
>     await StatusBar.setOverlaysWebView({ overlay: true });
> }, 500);

Yes after a ton of googling and whatnot this is what i've had to resort to aswell, my issue was with StatusBar.setBackgroundColor not setting the correct statusbar color on app launch (the statusbar background is supposed to be white or black depending on whether the user last chose dark or light theme for the app, and when the user toggles themes in the app the statusbar background color changes between white and black just fine), putting a settimeout to occur 250ms after the splashscreen is hidden did the trick, feels a bit hacky and i was still looking for a proper fix but it doesn't look like there is one so this will have to do.

**Comment by sam-higgs** [comment]
Still an issue for me with version 5.0.6

**Comment by aacassandra** [comment]
tested on ionic/vue very clean
```
  onMounted(() => {
    Device.getInfo().then((info) => {
      if (info.platform !== 'web') {
        SplashScreen.hide().then(() => {
          setTimeout(() => {
            StatusBar.setOverlaysWebView({ overlay: true })
            SafeArea.getStatusBarHeight().then((res) => {
              statusBarHeight.value = res.height + 3
            })
          }, 100)
        })
      }
    })
  })
  ```
  
  with capacitor.config.ts
  ```
  import { CapacitorConfig } from '@capacitor/cli'

const config: CapacitorConfig = {
  appId: 'io.ionic.starter',
  appName: 'E-Kampoeng',
  webDir: 'dist',
  server: {
    androidScheme: 'https',
  },
  plugins: {
    SplashScreen: {
      launchShowDuration: 30000,
      // launchAutoHide: true,
      launchFadeOutDuration: 700,
      backgroundColor: '#ffffffff',
      // androidSplashResourceName: 'splash',
      // androidScaleType: 'CENTER_CROP',
      showSpinner: false,
      androidSpinnerStyle: 'small',
      iosSpinnerStyle: 'small',
      spinnerColor: '#999999',
      splashFullScreen: true,
      splashImmersive: true,
      layoutName: 'launch_screen',
      // useDialog: true,
    },
  },
}

export default config

  ```

**Comment by haily2706** [comment]
Still an issue for me with version 5.0.6, react/ionic - I tried all proposed approach above? it did not work with my project. Has anyone solved this issue? please give me a hint to overcome it or when this issue will be fixed. Thank you!

**Comment by prdjed** [comment]
Using Angular here, best solution I found is to set splashscreen duration to some reasonable time:
`SplashScreen:{    
      launchShowDuration:2300
  },`

and in my app.component.ts:

`this.platform.ready().then(()=>{
     
     setTimeout(async()=>{
      await StatusBar.setOverlaysWebView({ overlay: true });
    },2300)
    })`
    
it works every time. And it is better than doing it with this approach:

`await SplashScreen.hide();
setTimeout(async () => {
    await StatusBar.setOverlaysWebView({ overlay: true });
}, 500);`
,because this way you will have blank white screen for short amount of time between splashscreen and app home screen.

**Comment by Dayonel** [comment]
> @nikitalpopov Thanks for the reply, but it's not working in my project.
> 
> I have been suggested to make changes to the status bar in a `setTimeout` after hiding the splash screen. I don't like the fix, but at least it's working in my case:
> 
> ```
> await SplashScreen.hide();
> setTimeout(async () => {
>     await StatusBar.setOverlaysWebView({ overlay: true });
> }, 500);
> ```

thank you so much!!

**Comment by mleister97** [comment]
The issue still does exist, all methods of `StatusBar` are affected.

**Comment by dimeloper** [comment]
Just FYI the issue still exists on `"@capacitor/status-bar": "6.0.0"`. It took me 2 days to find this github issue and try out the suggested timeout, equal to the `launchShowDuration` that finally makes it work for me too. Maybe until a proper fix is introduced this "workaround" should also be included in the documentation of the [status-bar plugin](https://capacitorjs.com/docs/apis/status-bar).

**Comment by Mimateam** [comment]
The issue still exists. Currently, the only solution is to use setTimeout. I am using "@capacitor/status-bar": "^5.0.8".

**Comment by Timeswitch** [comment]
Same issue, struggled to get the Splashscreen only to deal with this :)

**Comment by aeharding** [comment]
I was just bit by this issue in prod :'(

Simply installing `@capacitor/splash-screen` in your project breaks status bar immersive mode. Capacitor just completely clobbers and breaks it.

**Comment by eric-horodyski** [comment]
This thread has been inactive for some time, and we’re cleaning up older items to keep the backlog focused and manageable.

Details on the effort:
[https://ionic.io/blog/keeping-capacitors-backlog-healthy](https://ionic.io/blog/keeping-capacitors-backlog-healthy)

If this is still relevant today, feel free to open a new issue with current information and we’ll review it.
